osu.Game/Online/API/Requests/GetScoresRequest.cs

```diff
@@ -16,6 +16,19 @@ public class GetScoresRequest : APIRequest<GetScoresResponse>
         public GetScoresRequest(BeatmapInfo beatmap)
         {
             this.beatmap = beatmap;
+
+            Success += onSuccess;
+        }
+
+        private void onSuccess(GetScoresResponse r)
+        {
+            foreach (OnlineScore score in r.Scores)
+            {
+                score.Beatmap = beatmap;
+                score.Ruleset = beatmap.Ruleset;
+
+                score.ResolveModString();

```

Comment:
i'd just make this 
`score.ApplyBeatmap(beatmap);`
and do everything inside there.

Also `OnlineScore` can probably be a nested private class inside `GetScoresResponse`. Nothing else should need to know about it, right?

Reply:
GetScoresResponse needs the OnlineScore class, too. Therefore it has to be public.

Reply:
Why? Can't they just be `Score`s?

Reply:
The JsonProperties are now part of OnlineScore so it has to be used for the parsing, right?

Reply:
hmm, right. i'd still consider making it nested to ensure it's only used here (can stay `public`, just shouldn't be referenced elsewhere)